### ISSUE TITLE
Responsive Broken

### ISSUE THREAD

**Issue by AustinMReppert** [iced-rs/iced#1879]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

Responsive is broken. The size provided in the closure will update, but the components inside will not update.

https://github.com/iced-rs/iced/blob/fcb1b454368638209862aeb5db41bc5f7d6d51a7/widget/src/lazy/responsive.rs#LL63C35-L63C35
This condition was added to only update the content layout if it is invalidated, however the content layout does not invalidate itself. 

### What is the expected behavior?

The content layout should be invalidated? Or we need to set the layout to None in update?
```rust
if self.size == new_size {
    return;
}
self.layout = None; // ?
```

### Version

master

### Operative System

Windows

**Comment by hecrj** [comment]
Do you have an SSCCE?

**Comment by AustinMReppert** [comment]
```rust
use iced::advanced::Widget;
use iced::widget::{Container, Responsive};
use iced::{Background, Color, Element, Sandbox, Settings};

pub fn main() -> iced::Result {
    HelloWorld::run(Settings::default())
}

#[derive(Debug, Clone, Copy)]
pub enum Message {}

struct HelloWorld {}

impl Sandbox for HelloWorld {
    type Message = Message;

    fn new() -> Self {
        HelloWorld {}
    }

    fn title(&self) -> String {
        "Test".to_string()
    }

    fn update(&mut self, message: Self::Message) {}

    fn view(&self) -> Element<Message> {
        Responsive::new(|view| {
            Container::new("text")
                .width(view.width / 2.0)
                .height(view.height / 2.0)
                .style(|theme: &iced::Theme| iced::widget::container::Appearance {
                    text_color: None,
                    background: Some(Background::Color(Color::new(1.0, 0.0, 0.0, 1.0))),
                    border_radius: Default::default(),
                    border_width: 0.0,
                    border_color: Default::default(),
                })
                .into()
        })
        .into()
    }
}
```

**Comment by hecrj** [comment]
Thanks! Should be fixed by #1890.
